src/ralph/backends/http/lrs.py

```diff
@@ -0,0 +1,89 @@
+"""LRS HTTP backend for Ralph."""
+
+import logging
+from typing import Iterable, Union
+from urllib.parse import urlparse
+from more_itertools import chunked
+
+from httpx import Client, Headers, HTTPStatusError, RequestError
+from pydantic import HttpUrl
+
+from ralph.conf import settings
+from ralph.exceptions import BackendException
+
+from .base import BaseHTTP
+
+lrs_settings = settings.BACKENDS.HTTP.LRS
+logger = logging.getLogger(__name__)
+
+
+class LRSHTTP(BaseHTTP):
+    """LRS HTTP backend."""
+
+    name = "lrs"
+
+    def __init__(
+        self,
+        connection_url: HttpUrl = lrs_settings.CONNECTION_URL,
+        headers: Headers = None,
+    ):
+        """Instantiates the LRS client.
+
+        Args:
+            connection_url (HttpUrl): Connection URL to a LRS.
+            headers (json): Headers defined for the LRS server connection
+        """
+
+        self.auth = (connection_url.user, connection_url.password)
+        self.headers = headers
+
+    def get(self, target: str, chunk_size: Union[None, int] = None):
+        """Get statements from LRS.
+
+        The `get` method defined in the LRS spefication returns `statements` array and
+        `more` IRL. The latter is required when the `statement` list returned has been
+        limited.
+        """
+
+        origin = urlparse(target).scheme + "://" + urlparse(target).netloc
+
+        def fetch_statements(url, origin=origin):
+            with Client(auth=self.auth, headers=self.headers) as client:
```

Comment:
This part (`Client(auth=self.auth, headers=self.headers)`) will return an error if `self.auth == (None, None)` which would be the case if no `user:password` is not provided in the `connection_url`.

Perhaps this should be a bit more generic ? I am not sure the `connection_url` in the class instantiation is necessary.